src/EVM/Solvers.hs

```diff
@@ -69,6 +71,9 @@ data MultiSol = MultiSol
 -- | A script to be executed, a list of models to be extracted in the case of a sat result, and a channel where the result should be written
 data Task = TaskSingle SingleData | TaskMulti MultiData
 
+newtype UnsatCache = UnsatCache [Prop]
```

Comment:
I think this should be renamed, no? `[Prop]` is not the unsat cache. It is more like an `Entry` that should be added to unsat cache.

Reply:
Good point! Renamed it `CacheEntry`, I hope that works :)